pytorch_lightning/callbacks/timer.py

```diff
@@ -0,0 +1,132 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Timer
+^^^^^
+"""
+import logging
+from datetime import datetime, timedelta
+from typing import Any, Dict, Union
+
+from pytorch_lightning.callbacks.base import Callback
+from pytorch_lightning.utilities import LightningEnum
+from pytorch_lightning.utilities.distributed import rank_zero_info
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+log = logging.getLogger(__name__)
+
+
+class Interval(LightningEnum):
+    step = "step"
+    epoch = "epoch"
+
+
+class Timer(Callback):
+    """
+    The Timer callback tracks the time spent in the training loop and interrupts the Trainer
+    if the given time limit is reached.
+
+    Args:
+        duration: A string in the format DD:HH:MM:SS (days, hours, minutes seconds), or a :class:`datetime.timedelta`.
+        interval: Determines if the interruption happens on epoch level or mid-epoch.
+            Can be either `epoch` or `step`.
+        verbose: Set this to ``False`` to suppress logging messages.
+
+    Raises:
+        MisconfigurationException:
+            If ``interval`` is not one of the supported choices.
+
+    Example::
+        from pytorch_lightning import Trainer
+        from pytorch_lightning.callbacks import Timer
+
+        # stop training after 12 hours
+        timer = Timer(duration="00:12:00:00")
+
+        # or provide a datetime.timedelta
+        from datetime import timedelta
+        timer = Timer(duration=timedelta(weeks=1))
+
+        # or provide a dictionary
+        timer = Timer(duration=dict(weeks=4, days=2))
+
+        # force training to stop after given time limit
+        trainer = Trainer(callbacks=[timer])
+    """
+
+    def __init__(
+        self,
+        duration: Union[str, timedelta, Dict[str, int]],
+        interval: str = Interval.step,
+        verbose: bool = True,
+    ):
+        super().__init__()
+        if isinstance(duration, str):
+            dhms = duration.strip().split(":")
+            dhms = [int(i) for i in dhms]
+            duration = timedelta(days=dhms[0], hours=dhms[1], minutes=dhms[2], seconds=dhms[3])
+        if isinstance(duration, dict):
+            duration = timedelta(**duration)
+        if interval not in set(Interval):
```

Comment:
I believe this works

```suggestion
        if interval not in Interval:
```

Reply:
```python
cls = <enum 'Interval'>, member = 'step'

    def __contains__(cls, member):
        if not isinstance(member, Enum):
>           raise TypeError(
                "unsupported operand type(s) for 'in': '%s' and '%s'" % (
                    type(member).__qualname__, cls.__class__.__qualname__))
E           TypeError: unsupported operand type(s) for 'in': 'str' and 'EnumMeta'
```
unfortunately not :) 
maybe something we can add to LightningEnum